Stream analytics output to CosmosDB (#16441)
hashicorp/terraform-provider-azurerm: website/docs/r/stream_analytics_output_cosmosdb.html.markdown

  name                = "cosmos-sql-db"
  resource_group_name = azurerm_cosmosdb_account.example.resource_group_name
  account_name        = azurerm_cosmosdb_account.example.name
  throughput          = 400
}

resource "azurerm_cosmosdb_sql_container" "example" {
  name                = "examplecontainer"
  resource_group_name = azurerm_cosmosdb_account.example.resource_group_name
  account_name        = azurerm_cosmosdb_account.example.name
  database_name       = azurerm_cosmosdb_sql_database.example.name
  partition_key_path  = "foo"
}

resource "azurerm_stream_analytics_output_cosmosdb" "example" {
  name                      = "output-to-cosmosdb"
  stream_analytics_job_name = azurerm_stream_analytics_job.example.name
  resource_group_name       = azurerm_stream_analytics_job.example.resource_group_name
  account_id                = azurerm_cosmosdb_account.example.name
  account_key               = azurerm_cosmosdb_account.example.primary_access_key
  database                  = azurerm_cosmosdb_sql_database.example.name
  collection_name_pattern   = azurerm_cosmosdb_sql_container.example.id
}
```

## Arguments Reference

The following arguments are supported:

> * `name` - (Required) The name of the Stream Output. Changing this forces a new resource to be created.

Reviewer: ```suggestion
* `name` - (Required) The name of the Stream Analytics Output. Changing this forces a new resource to be created.
```

> * `resource_group_name` - (Required) The name of the Resource Group where the Stream Analytics Job exists. Changing this forces a new resource to be created.
> 
> * `stream_analytics_job_name` - (Required) The name of the Stream Analytics Job. Changing this forces a new resource to be created.

Reviewer: ```suggestion
* `stream_analytics_job_id` - (Required) The ID of the Stream Analytics Job. Changing this forces a new resource to be created.
```

Author: Will change.

> * `account_name` - (Required) The name of the CosmosDB Account.

Reviewer: ```suggestion
* `account_name` - (Required) The name of the CosmosDB account.
```

> * `account_key` - (Required) The Account Key for the CosmosDB database.

Reviewer: ```suggestion
* `account_key` - (Required) The account key for the CosmosDB database.
```

> * `database` - (Required) The name or id of the CosmosDB database.

Reviewer: As per the validation set for the property we only accept the ID.
```suggestion
* `cosmosdb_sql_database_id` - (Required) The ID of the CosmosDB database.
```

> * `collection_name_pattern` - (Required) The collection name pattern for the collection to be used. It can be the name of the container, or a collection name format which contains an optional {partition} token.

Reviewer: As mentioned above I think it makes more sense for this to be named `container_id` - I also haven't seen an a test that provides this property with an optional partition token. Could you find an example of what that looks like? This might need to be added as an additional/separate field to the resource.

* `document_id` - (Optional) The name of the field in output events used to specify the primary key which insert or update operations are based on.

> * `partition_key` - (Optional) The name of the field in output events used to specify the key for partitioning output across collections. If 'collection_name_pattern' contains the {partition} token, this property is required to be specified.

Reviewer: If this is required in specific circumstances should we be checking this in the create function?

Author: If the `partition_key` does not exist under the described circumstance, an error message will show up to complain about the missing field, test proven.

## Attributes Reference

In addition to the Arguments listed above - the following Attributes are exported:

> * `id` - The ID of the Stream Analytics Output CosmosDB.

Reviewer: ```suggestion
* `id` - The ID of the Stream Analytics Output for CosmosDB.
```

## Timeouts

The `timeouts` block allows you to specify [timeouts](https://www.terraform.io/docs/configuration/resources.html#timeouts) for certain actions:

> * `create` - (Defaults to 30 minutes) Used when creating the Stream Analytics.
> * `read` - (Defaults to 5 minutes) Used when retrieving the Stream Analytics.
> * `update` - (Defaults to 30 minutes) Used when updating the Stream Analytics.
> * `delete` - (Defaults to 30 minutes) Used when deleting the Stream Analytics.

Reviewer: ```suggestion
* `create` - (Defaults to 30 minutes) Used when creating the Stream Analytics Output for CosmosDB.
* `read` - (Defaults to 5 minutes) Used when retrieving the Stream Analytics Output for CosmosDB.
* `update` - (Defaults to 30 minutes) Used when updating the Stream Analytics Output for CosmosDB.
* `delete` - (Defaults to 30 minutes) Used when deleting the Stream Analytics Output for CosmosDB.
```

## Import

> Stream Analytics Output to CosmosDB can be imported using the `resource id`, e.g.

Reviewer: ```suggestion
Stream Analytics Outputs for CosmosDB can be imported using the `resource id`, e.g.
```
